pkgs/development/libraries/protobuf/generic-v3.nix

```diff
@@ -31,10 +30,10 @@ stdenv.mkDerivation rec {
   '';
 
   nativeBuildInputs = [ autoreconfHook buildPackages.which buildPackages.stdenv.cc ]
-                      ++ stdenv.lib.optional useExternalProtoc [ buildProtobuf ];
+                      ++ (if buildProtobuf == null then [] else [ buildProtobuf ]);
```

Comment:
`buildInputs` entries which are null will just be ignored, so you don't need to put this condition here.